packages/sveltekit/test/server/handle.test.ts

```diff
@@ -0,0 +1,263 @@
+    captureException: (err: unknown, cb: (arg0: unknown) => unknown) => {
+      cb(mockScope);
+      mockCaptureException(err, cb);
+      return original.captureException(err, cb);
+    },
+  };
+});
+
+const mockAddExceptionMechanism = vi.fn();
+
+vi.mock('@sentry/utils', async () => {
+  const original = (await vi.importActual('@sentry/utils')) as any;
+  return {
+    ...original,
+    addExceptionMechanism: (...args: unknown[]) => mockAddExceptionMechanism(...args),
+  };
+});
+
+function mockEvent(override: Record<string, unknown> = {}): Parameters<Handle>[0]['event'] {
+  // From https://github.com/sveltejs/kit/blob/34fd2a5adfc4408984a4eeb5dda38d7073b587b8/packages/kit/src/runtime/server/respond.js#L124
+  const event: Parameters<Handle>[0]['event'] = {
+    cookies: {} as any,
+    fetch: () => Promise.resolve({} as any),
+    getClientAddress: () => '',
+    locals: {},
+    params: { id: '123' },
+    platform: {},
+    request: {
+      method: 'GET',
+      headers: {
+        get: () => null,
+        append: () => {},
+        delete: () => {},
+        forEach: () => {},
+        has: () => false,
+        set: () => {},
+      },
+    } as any,
+    route: { id: '/users/[id]' },
+    setHeaders: () => {},
+    url: new URL('http://localhost:3000/users/123'),
+    isDataRequest: false,
+
+    ...override,
+  };
+
+  return event;
+}
+
+const mockResponse = { status: 200, headers: {}, body: '' } as any;
+
+const enum Type {
+  Sync = 'sync',
+  Async = 'async',
+}
+
+function resolve(type: Type, isError: boolean): Parameters<Handle>[0]['resolve'] {
+  if (type === Type.Sync) {
+    return (..._args: unknown[]) => {
+      if (isError) {
+        throw new Error(type);
+      }
+
+      return mockResponse;
+    };
+  }
+
+  return (..._args: unknown[]) => {
+    return new Promise((resolve, reject) => {
+      if (isError) {
+        reject(new Error(type));
+      } else {
+        resolve(mockResponse);
+      }
+    });
+  };
+}
+
+let hub: Hub;
+let client: NodeClient;
+
+describe('handleSentry', () => {
+  beforeAll(() => {
+    addTracingExtensions();
+  });
+
+  beforeEach(() => {
+    mockScope = new Scope();
+    const options = getDefaultNodeClientOptions({ tracesSampleRate: 1.0 });
+    client = new NodeClient(options);
+    hub = new Hub(client);
+    makeMain(hub);
+
+    mockCaptureException.mockClear();
+    mockAddExceptionMechanism.mockClear();
+  });
+
+  describe.each([
+    // isSync, isError
+    [Type.Sync, true],
+    [Type.Sync, false],
+    [Type.Async, true],
+    [Type.Async, false],
+  ])('%s resolve with error %s', (type, isError) => {
+    it('should return a response', async () => {
+      expect.assertions(isError ? 2 : 1);
+
+      let response;
+      try {
+        response = await sentryHandle({ event: mockEvent(), resolve: resolve(type, isError) });
+      } catch (e) {
+        expect(e).toBeInstanceOf(Error);
+        expect(e.message).toEqual(type);
+      }
+
+      if (!isError) {
```

Comment:
m: hmm not really a fan of conditional paths in tests. Makes tests harder to understand and also feels like we're testing different scenarios in one test. To me (maybe just personal preference), it seems like we're misusing `each` here a little. That being said, if we split the `isError` cases to separate tests, we probably end up with a ton of duplicated code, right? 

I'll leave this up to you: If refactoring these tests results in too much duplication or work, I'm fine with leaving it as is. Generally though, let's try to avoid conditional expects and test paths.

Reply:
Yeah I struggled with this a little, but one of the issues here is that there are 4 different scenarios we need to account for.

1. Async and no error
2. Async and throw an error
3. Sync and no error
4. Sync and throws no error

Let me try cleaning it up so we can remove the `isError` checks.

Reply:
[376291d](https://github.com/getsentry/sentry-javascript/pull/7532/commits/376291d88d07ea617dafd6ccecb137fa2463cd59)